apl-tools/src/main/java/com/apollocurrency/aplwallet/apl/tools/impl/heightmon/HeightMonitorServiceImpl.java

```diff
@@ -0,0 +1,270 @@
+    public void init() {
+        client = createHttpClient();
+        try {
+            client.start();
+            executor = Executors.newFixedThreadPool(30);
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e.toString(), e);
+        }
+    }
+
+    @Override
+    public void setUp(HeightMonitorConfig config) {
+        PeersConfig peersConfig = config.getPeersConfig();
+        this.port = peersConfig.getDefaultPort();
+        this.peers = Collections.synchronizedList(peersConfig.getPeersInfo().stream().peek(this::setDefaultPortIfNull).collect(Collectors.toList()));
+        this.maxBlocksDiffCounters = createMaxBlocksDiffCounters(config.getMaxBlocksDiffPeriods() == null ? DEFAULT_PERIODS : config.getMaxBlocksDiffPeriods());
+        this.peerApiUrls = this.peers.stream().map(this::createUrl).collect(Collectors.toList());
+
+    }
+
+    private PeerInfo setDefaultPortIfNull(PeerInfo peerInfo) {
+        if (peerInfo.getPort() == null) {
+            peerInfo.setPort(port);
+        }
+        return peerInfo;
+    }
+
+    @Override
+    public boolean addPeer(PeerInfo peer) {
+        Objects.requireNonNull(peer);
+        setDefaultPortIfNull(peer);
+        String url = createUrl(peer);
+        boolean result = false;
+        if (!peers.contains(peer)) {
+            result = true;
+            peers.add(peer);
+            peerApiUrls.add(url);
+        }
+        return result;
+    }
+
+    private String createUrl(PeerInfo peerInfo) {
+        return String.format(URL_FORMAT, peerInfo.getSchema(), peerInfo.getHost(), peerInfo.getPort());
+    }
+
+
+    @PreDestroy
+    public void shutdown() {
+        try {
+            client.stop();
+            executor.shutdownNow();
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e.toString(), e);
+        }
+    }
+
+    private HttpClient createHttpClient() {
+        SslContextFactory sslContextFactory = new SslContextFactory();
+        sslContextFactory.setTrustAll(true);
+        HttpClient httpClient = new HttpClient(sslContextFactory);
+        httpClient.setIdleTimeout(IDLE_TIMEOUT);
+        httpClient.setConnectTimeout(CONNECT_TIMEOUT);
+        httpClient.setFollowRedirects(false);
+        return httpClient;
+    }
+
+    private List<MaxBlocksDiffCounter> createMaxBlocksDiffCounters(List<Integer> maxBlocksDiffPeriods) {
+        return maxBlocksDiffPeriods.stream().map(MaxBlocksDiffCounter::new).collect(Collectors.toList());
+    }
+
+    public NetworkStats getLastStats() {
+        return lastStats.get();
+    }
+
+    @Override
+    public NetworkStats updateStats() {
+        log.info("===========================================");
+        Map<String, List<Block>> peerBlocks = getPeersBlocks();
+        log.info(String.format("%5.5s %5.5s %-16.16s %-16.16s %9.9s %7.7s %7.7s", "diff1", "diff2", "peer1", "peer2", "milestone", "height1", "height2"));
+        int currentMaxBlocksDiff = -1;
```

Comment:
Could you pls add 'VERSION' column here (and into JSON)? Sometime it's very useful.

Reply:
Added